google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java

```diff
@@ -338,18 +327,20 @@ public void testAutodetectWhenNotRequesterPays() throws IOException {
         "");
   }
 
+  @Test
+  public void testAutoDetectNoUserProject() throws IOException {
+    CloudStorageFileSystem testBucket = getRequesterPaysBucket(false, "");
```

Comment:
To avoid dealing with [this](https://github.com/googleapis/java-storage-nio/blob/4597b14fad27a9d5940609e7a2942f02b78830ef/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java#L206) check, we set autodetect to false and then just check the provider value for requesterPays directly.